test/icon.c

```diff
@@ -14,6 +14,55 @@
 #define IS_ICON_PNG(pb)  4 == gdk_pixbuf_get_width(pb)
 #define IS_ICON_SVG(pb) 16 == gdk_pixbuf_get_width(pb)
 
+static const char *skipchars(const char *string, size_t chars)
+{
+        g_assert(chars <= strnlen(string, chars));
+        return string+chars;
+}
+
+TEST test_get_pixbuf_from_file_tilde(void)
+{
+        char *cwd  = g_get_current_dir();
+        const char *home = g_get_home_dir();
+        const char *iconpath = skipchars(ICONPREFIX, 1);
```

Comment:
Maybe instead of having this function to hack around literal strings set `ICONPREFIX` to `data/icons/path` and explicitly add the current directory prefix in the suite declaration.

Reply:
Valid point. Changed.

While writing the tests, I wasn't 100% happy about the code either. Thank you for those "nitpicks" I love them and I love to work with you together.